ui/src/components/WebRTCVideo.tsx

```diff
@@ -346,14 +395,125 @@ export default function WebRTCVideo() {
       setIsCapsLockActive(e.getModifierState("CapsLock"));
       setIsScrollLockActive(e.getModifierState("ScrollLock"));
 
+      // Check if the released key is a modifier (e.g., Shift, Alt, Control)
+      const isModifierKey =
+      e.code === "ShiftLeft" ||
+      e.code === "ShiftRight" ||
+      e.code === "AltLeft" ||
+      e.code === "AltRight" ||
+      e.code === "ControlLeft" ||
+      e.code === "ControlRight";
+
+      var newKeys = prev.activeKeys;
+
+      // Handle modifier release
+      if (isModifierKey) {
+        // Update all affected keys when this modifier is released
+        activeKeyState.current.forEach((value, code) => {
+          const { mappedKey, modifiers: mappedModifiers} = value;
+
+          // Remove the released modifier from the modifier bitmask
+          //const updatedModifiers = modifiers & ~modifiers[e.code];
+
+          // Recalculate the remapped key based on the updated modifiers
+          //const updatedMappedKey = chars[originalKey]?.key || originalKey;
+
+          var removeCurrentKey = false;
+
+          // Shift Handling
+          if (mappedModifiers.shift && (e.code === "ShiftLeft" || e.code === "ShiftRight")) {
```

Comment:
Doesn't this conflate left and right shift keys such that if I press left,, hold, press right, release (left or right), then we still have a shift key down, but the shift state says we are *not shifted*... so we're wrong... and then we release the other (right or left) and now we have no physical shift keys down, but the shift state is now *shifted*... so we're even wronger... \